lib/src/node.c

```diff
@@ -542,7 +542,8 @@ TSNode ts_node_parent(TSNode self) {
 
   while (true) {
     TSNode next_node = ts_node_child_containing_descendant(node, self);
-    if (ts_node_is_null(next_node)) break;
+    if (next_node.id == self.id
```

Comment:
I think this is a bug in `ts_node_child_containing_descendant()`, since it's not supposed to return the node itself. 
It has also been deprecated in favor of `ts_node_child_with_descendant()`. I think the updated version would look something like this:

```c
TSNode next_node = ts_node_child_with_descendant(node, self);
if (ts_node_is_null(next_node)) return ts_node__null();
if (next_node.id == self.id) break;
```

Reply:
Good point. However, using that exact code causes the library to hang when running the test suite. Just swapping `ts_node_child_containing_descendant` for `ts_node_child_with_descendant` seems to work though:

```c
TSNode next_node = ts_node_child_with_descendant(node, self);
if (next_node.id == self.id
    || ts_node_is_null(next_node)) break;
node = next_node;
```